server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -2725,13 +2726,22 @@ public long getMaxSeqNoOfUpdatesOrDeletes() {
     }
 
     /**
-     * Advances the max_seq_no_of_updates marker of the engine of this shard to at least the given sequence number.
+     * A replica calls this method to advance the max_seq_no_of_updates marker of its engine to at least the max_seq_no_of_updates
+     * value (piggybacked in a replication request) that it receives from its primary before executing that replication request.
+     * The receiving value is at least the highest max_seq_no_of_updates of all index/delete operations in that replication request.
```

Comment:
it's not clear here what "msu of all" means (it's clear in the google doc we have). Maybe write that value is at least as high as the msu on the primary was when any of the operations were processed on it?